pkgs/build-support/fetchtorrent/tests.nix

```diff
@@ -2,90 +2,130 @@
+  # supported by fetchtorrent appear to support torrents that are only reliably
+  # seeded by HTTP sources rather than other people using Bittorrent clients.
+  # Sintel was the smallest torrent I could find that had a free license and
+  # was reliably seeded by other Bittorrent clients.
+  #
+  # For more information, see the discussion at
+  # https://github.com/NixOS/nixpkgs/pull/432091/files/bd13421b2b70f3f125061018c800439ef2d43e8d#r2264073113
   sintel.meta = {
     description = "An open source short film to show off open source technologies.";
     longDescription = ''
       An independently produced short film, initiated by the Blender Foundation
-      as a means to further improve andvalidate the free/open source 3D
+      as a means to further improve and validate the free/open source 3D
       creation suite Blender.
     '';
     license = lib.licenses.cc-by-30;
     homepage = "https://durian.blender.org/";
   };
 
   # Via https://webtorrent.io/free-torrents
-  http.url = "https://webtorrent.io/torrents/sintel.torrent";
+  http.url = "${./test-sintel.torrent}";
   magnet.url = "magnet:?xt=urn:btih:08ada5a7a6183aae1e09d831df6748d566095a10&dn=Sintel&tr=udp%3A%2F%2Fexplodie.org%3A6969&tr=udp%3A%2F%2Ftracker.coppersurfer.tk%3A6969&tr=udp%3A%2F%2Ftracker.empire-js.us%3A1337&tr=udp%3A%2F%2Ftracker.leechers-paradise.org%3A6969&tr=udp%3A%2F%2Ftracker.opentrackr.org%3A1337&tr=wss%3A%2F%2Ftracker.btorrent.xyz&tr=wss%3A%2F%2Ftracker.fastcast.nz&tr=wss%3A%2F%2Ftracker.openwebtorrent.com&ws=https%3A%2F%2Fwebtorrent.io%2Ftorrents%2F&xs=https%3A%2F%2Fwebtorrent.io%2Ftorrents%2Fsintel.torrent";
 
-  flattened.hash = "sha256-EzbmBiTEWOlFUNaV5R4eDeD9EBbp6d93rfby88ACg0s=";
-  unflattened.hash = "sha256-lVrlo1AwmFcxwsIsY976VYqb3hAprFH1xWYdmlTuw0U=";
+  # Sintel isn't a massive download, but it's not small.  There's also no real
+  # value in storing copies of it in Nix caches, which is what happens by
+  # default when this test succeeds.  Avoid that by verifying the downloaded
+  # files using `sha512sum` in the post-fetch hook, then deleting the files so
+  # the actual derivation result is an empty directory.
+  #
+  # Chain `&&` in the postFetch phase because the transmission backend does not
+  # run that phase with `errexit` enabled.
+  flattened.postFetch = ''
+    pushd "$out" &&
+    sha512sum --check --strict ${./test-hashes.sha512sum} &&
+    sed 's/.*  //' ${./test-hashes.sha512sum} | xargs rm --verbose &&
+    popd
+  '';
+  unflattened.postFetch = ''
+    pushd "$out" &&
+    pushd Sintel &&
+    sha512sum --check --strict ${./test-hashes.sha512sum} &&
+    sed 's/.*  //' ${./test-hashes.sha512sum} | xargs rm --verbose &&
+    popd &&
+    rm --dir --verbose Sintel &&
+    popd
+  '';
+
+  # Fixed output derivation hash is identical for all derivations: the empty
+  # directory.
+  fetchtorrentWithHash =
+    args:
+    fetchtorrent (
+      {
+        hash = builtins.convertHash {
```

Comment:
Isn't the minimum Nix version to evaluate nixpkgs 2.18?

I'm pretty sure that this builtin was added later considering that e.g. Lix doesn't have it.